src/services/TableService.js

```diff
@@ -5,123 +5,166 @@ angular.module('mms')
+     
 
-        // rotate row-based 2D array into col-based 2D array
-        ids = _.unzip(data.body).map(function(col) {
-          return col.map(function(cell) {
-            // extract only the SysML ID
-            return cell.content[0].source;
-          });
+      ElementService.getElements(dataValuesMmmEid, 1)
+        .then(function(refValues) {
+        var counter = 0;
+        var values = [];
+        dataValuesMmmEid.elementIds.forEach(function(elementId){
+          if (elementId.indexOf("_") === 0) {//md id
+            values.push(refValues[counter++]);
+          }
+          else {
+            values.push(elementId);
+          }
         });
+        var dataTableValues = [];
+        var datavalues = [];
+        var startIndex = 0;
+        var dataIdFilters=[];
+        counter = 0; //reset
+        var valueLength = numOfDataColumn* numOfRowHeadersPerTable;//rowHeadersMmsEid.length;
+        for (i = 0; i < valueLength; i= i + numOfDataColumn){
+          var datarow =[];// new Array(tableColumnHeadersLabels[k].length);
+          for (j = 0; j < numOfDataColumn; j++){
+            datarow.push(values[counter++]); 
+          }
+          datavalues.push(datarow);
+        }
 
-        var promises = [];
-        ids.forEach(function(col, i) {
-          columns[i] = [];
-          promises.push(ElementService.getElements(col, false, ws, version)
-          .then(function(data) {
-            data.forEach(function(cell) {
-              var val;
-              // LiteralReal => double
-              // Number, integer, etc.?
-              if (typeof cell.specialization.value !== 'undefined') {
-                if (typeof cell.specialization.value[0].double !== 'undefined') {
-                  val = cell.specialization.value[0].double;
-                } else if (typeof cell.specialization.value[0].integer !== 'undefined') {
-                  val = cell.specialization.value[0].integer;
-                } else if (typeof cell.specialization.value[0].expressionBody !== 'undefined') {
-                  val = cell.specialization.value[0].expressionBody[0];
-                } else if (typeof cell.specialization.value[0].string !== 'undefined') {
-                  val = cell.specialization.value[0].string;
-                }
-                if (val !== undefined) {
-                  if (keys[i] === undefined) {
-                    keys[i] = cell.name;
-                  }
-                  columns[i].push(val);
-                } else {
-                  // unknown value
-                  console.log('No value found:');
-                  console.log(cell);
-                }
-              }
-            });
-          }));
-        });
+        var filters = [];
+        var filterRowHeaders = [];
+        var filterColumnHeaders=[];
+        for ( i = 0; i < tableRowHeaders.length; i++){
+          if ( tableRowHeaders[i].type === "InstanceSpecification")
```

Comment:
instead of trying to differentiate by the element type, it's better to check what it's actually displaying (based on the sourceProperty or sourceType from the table spec), that means you'll need to keep track of that somewhere